addons/portal/static/src/js/portal_composer.js

```diff
@@ -138,15 +137,38 @@ var PortalComposer = publicWidget.Widget.extend({
         });
     },
     /**
-     * Returns a Promise that is never resolved to prevent sending the form
-     * twice when clicking twice on the button, in combination with the `async`
-     * in the event definition.
+     * prepares data to send message
      *
      * @private
+     */
+    _prepareMessageData: function () {
+        return _.extend(this.options || {}, {
+            'message': this.$('textarea[name="message"]').val(),
+            'attachment_ids': _.pluck(this.attachments, 'id'),
+            'attachment_tokens': _.pluck(this.attachments, 'access_token'),
+        });
+    },
+    /**
+     * Send message using rpc call and display new message and message count
+     *
+     * @private
+     * @param {Event} ev
      * @returns {Promise}
      */
-    _onSubmitButtonClick: function () {
-        return new Promise(function (resolve, reject) {});
+    _onSubmitButtonClick: async function (ev) {
+        ev.preventDefault();
+        if (!this.$inputTextarea.val().trim() && !this.attachments.length) {
+            this.$inputTextarea.addClass('border-danger');
+            let error = _t('Oops! Some fields are required. Please make sure that message or attachment is not empty!');
+            this.$(".o_portal_chatter_composer_error").text(error).removeClass('d-none');
+            return;
+        }
+        const result = await this._rpc({
+            route: ev.currentTarget.getAttribute('data-action'),
+            params: this._prepareMessageData(),
+        });
+        core.bus.trigger('reload_chatter_content', result);
+        return result;
```

Comment:
Not sure why you're returning anything here. This is an event handler, nothing is using that result.

Reply:
we need to return the data as result to overrode _onSubmitButtonClick of portal_rating and handle there or simple  return; (if no message).  otherwise, we will get result undefined in both cases in the overrode method...

Reply:
Didn't catch that. A small comment here to explain this unusual behaviour would be nice :+1:

Reply:
@sdegueldre  we suggest to add a private method in portal's portal_composer.js. we will call click event (_onSubmitButtonClick) in this method.. and override new method in portal_rating's portal_composer.js.. to avoid unusual behaviour on the event handler method.. in this case we would not have to return data from click event.. what is your thought?